src/openzaak/components/zaken/api/serializers/zaken.py

```diff
@@ -786,8 +797,38 @@ def create(self, validated_data):
             # Save updated information on the ZAAK
             zaak.save(update_fields=_zaak_fields_changed)
 
+            # update deelzaken
+            self.update_deelzaken(
+                zaak.deelzaken,
+                archiefactiedatum=zaak.archiefactiedatum,
+                startdatum_bewaartermijn=zaak.startdatum_bewaartermijn,
+            )
+
         return obj
 
+    def update_deelzaken(self, qs, **fields):
+        self._update_deelzaken_with_internal_catalogi(
+            qs.filter(_zaaktype__isnull=False), **fields
+        )
+        self._update_deelzaken_with_external_catalogi(
+            qs.filter(_zaaktype_relative_url__isnull=False), **fields
+        )
+
+    def _update_deelzaken_with_internal_catalogi(self, qs, **fields):
+        qs.filter(
+            resultaat___resultaattype__brondatum_archiefprocedure_afleidingswijze=Afleidingswijze.hoofdzaak
+        ).update(**fields)
```

Comment:
Might be good to check this with Hugo, this also sets the `archiefactiedatum` equal to that of the Hoofdzaak, but the issue mentions `Bij alle Deelzaken die als Afleidingswijze "hoofdzaak" hebben, wordt de brondatum van de Hoofdzaak ingevuld.`, which would imply: `archiefactiedatum = brondatum_hoofdzaak + archiefactietermijn (of resultaattype for deelzaak)` (https://github.com/open-zaak/open-zaak/issues/1839#issuecomment-2696639374)

Reply:
let's verify this with Joeri during standup refinement

Reply:
What Steven mentions sounds valid so let's do that.